app/controllers/groups_controller.rb

```diff
@@ -457,6 +457,14 @@ def handle_membership_request
     user = User.find_by(id: params[:user_id])
     raise Discourse::InvalidParameters.new(:user_id) if user.blank?
 
+    # find original membership request PM
+    request_topic =
+      Topic.find_by(
+        title: I18n.t("groups.request_membership_pm.title", group_name: group.name),
```

Comment:
I guess we don't have another way to find the group membership request, other than the title? Asking because I see a possible edge case here, where the user can have a locale different from the site's default, and therefore the title in the message may be different. What should we do in that case?

Reply:
Thanks for the comment, @pmusaraj! Yeah, I also got a "code smell" from that and am not positive it's the most direct way to query for the original message 🤔 I'll play with that a bit.

In terms of the edge case you mentioned, my understanding is that as long as the `title` field matches the user's locale values in the translation ymls that this would work, since i18n is doing the heavy lifting there. I may be misunderstanding the edge case though? Perhaps you're asking about if the user's locale changed since the first message was sent?

Reply:
The edge case I am thinking is this: 

- as user1, I set my profile to Spanish
- I request permission to join group A
- Discourse creates a PM with my group membership request, but the title of the message is in Spanish
- admin sees that message in English, will the title be found when trying to reply?

Reply:
Ah, got it. Thanks for clarifying. Will work on this!

Reply:
Okay, I see the issue and was able to replicate in specs. To confirm your hypothesis, the title will _not_ be found. In the case where the users' locales don't match, should we defer back to the original method (two separate Topics), or is there another option you see working here? In the meantime, I'm going to poke around and look for a similar instance to see how that's being handled.

Reply:
It's also possible I can adjust that query to work regardless of translation issues. They all include the group name, so that's a starting point to work from 🕵️

Reply:
Alright, I think I've got this one fixed. In doing this, I was thinking we should probably send the acceptance message in the requesting user's language as well, so I changed that.